### PR TITLE
[UI] Created UI for treasury screen

### PR DESCRIPTION
**Pull request by Polymeth**
This PR adds the UI content for the treasury screen and it's corresponding tests. This includes :
- Main UI of the whole menu
- Placeholder list of current receipts
- Edit receipt screen

NB : I am a 3rd year student and neither my mom or dad is a senior google engineer so if you are concerned by my very poor utilization of git even though I try my best, directly DM me and talk to me.

### CODE REVIEW

**Review by Mai-LinhC — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/se/assocify/ui/screens/treasury/TreasuryScreen.kt [hunk 1]
```diff
@@ -1,35 +1,296 @@
...
+import com.github.se.assocify.ui.composables.MainNavigationBar
+import kotlinx.coroutines.launch
+
+// Number of tabs at the top
+const val NUMBER_OF_PAGES: Int = 3
+
+// Index of each tag for navigation
+const val PAGE_RECEIPT_INDEX: Int = 0
+const val PAGE_BUDGET_INDEX: Int = 1
+const val PAGE_BALANCE_INDEX: Int = 2
```
[the remark this answers is not in the document]
Comment (Polymeth) [reply app/src/main/java/com/github/se/assocify/ui/screens/treasury/TreasuryScreen.kt:60]: Good point right

File: app/src/main/java/com/github/se/assocify/ui/screens/treasury/TreasuryScreen.kt [hunk 2]
```diff
@@ -1,35 +1,296 @@
...
+                            .background(
+                                color = MaterialTheme.colorScheme.primary,
+                                shape = RoundedCornerShape(8.dp)))
+              }) {
+                TreasuryTab(
+                    selected = pagerState.currentPage == PAGE_RECEIPT_INDEX,
+                    onClick = {
+                      coroutineRoute.launch { pagerState.animateScrollToPage(PAGE_RECEIPT_INDEX) }
+                    },
+                    text = "My receipts",
```
[the remark this answers is not in the document]
Comment (Polymeth) [reply app/src/main/java/com/github/se/assocify/ui/screens/treasury/TreasuryScreen.kt:112]: I think for the moment we're gonna be conservative and change to "Receipts". Eventually, we'll change later right.

File: app/src/main/java/com/github/se/assocify/ui/screens/treasury/TreasuryScreen.kt [hunk 3]
```diff
@@ -1,35 +1,296 @@
...
+/**
+ * ------------------------------------------------- ** PAGES *
+ * * ------------------------------------------------- *
+ */
+
+/** My receipts UI page */
+@Composable
+private fun MyReceiptPage() {
+  // val receiptNames = listOf("Grocery Shopping", "Restaurant", "Gas Station", "Coffee Shop")
+  val receiptNames = List(50) { "Receipt $it" }
```
[the remark this answers is not in the document]
Comment (Polymeth) [reply app/src/main/java/com/github/se/assocify/ui/screens/treasury/TreasuryScreen.kt:153]: Will be subject to change in a future PR

File: app/src/main/java/com/github/se/assocify/ui/screens/treasury/TreasuryScreen.kt [hunk 4]
```diff
@@ -1,35 +1,296 @@
...
+                      lineHeight = 24.sp,
+                      color = Color(0xFF505050),
+                      letterSpacing = 0.sp,
+                  ))
+          Spacer(modifier = Modifier.width(8.dp))
+          Icon(
+              modifier = Modifier.size(20.dp).testTag("shoppingCartIcon"),
+              imageVector = Icons.Filled.ShoppingCart,
+              contentDescription = "Arrow icon",
+          )
```
[the remark this answers is not in the document]
Comment (Polymeth) [reply app/src/main/java/com/github/se/assocify/ui/screens/treasury/TreasuryScreen.kt:286]: Will depend on the viewmodel so will change on a later PR

File: app/src/androidTest/java/com/github/se/assocify/screens/TreasuryScreenTest.kt [hunk 5]
```diff
@@ -42,4 +43,29 @@ class TreasuryScreenTest : TestCase(kaspressoBuilder = Kaspresso.Builder.withCom
...
+      onNodeWithTag("balanceTab").assertIsSelected()
+
+      onNodeWithTag("myReceiptsTab").assertIsDisplayed()
+      onNodeWithTag("myReceiptsTab").performClick()
+      onNodeWithTag("myReceiptsTab").assertIsSelected()
+    }
+  }
+
+  @Test
+  fun createTodo() = run {
```
Comment (Mai-LinhC) [inline app/src/androidTest/java/com/github/se/assocify/screens/TreasuryScreenTest.kt:65]: will need to rename

**Review by Mai-LinhC — APPROVED** [review]

**Pull request merged**